### ISSUE TITLE
Training transformers with limited data

### ISSUE THREAD

**Issue by bwittmann** [himashi92/VT-UNet#4]
Hi @himashi92,
fist of all thank you very much for your great work!

I would have a question regarding how you coped with the issue of having only very limited training data available (MSD liver: 87 train samples). Transformer based architectures like ViT or also detectors like DeTr have shown to only perform well when there is huge amount of labeled data available (DeTr lower bound of 2D images ~15k to train from scratch). So I would think that training a 3D transformer based architecture like VT-UNet would even be more data hungry.

Have you made experiments without the pre-trained Swin-T weights as initialization? 
Did you use heavy data augmentation for those small data sets?
From which shape did you crop to 128x128x128?
Is it due to your shifted-window approach, that your architecture performs also very good with limited data / have you experimented with different transformer layers (like ViT)?
So the main question is basically: what are in your opinion the key-factors of the success of your approach when it comes to limited data?

I would be very happy if you could answer my questions.
BR Bastian

**Comment by himashi92** [comment]
Hi Bastian,

From our experiments, it can be seen for smaller datasets, training VT-UNet is harder and it behaves similar to other transformer architectures.

I have done an ablation study to show that there is an impact on performance when we use pre-trained weights in our pre-print. But that was performed only for the BraTS dataset. For all datasets, regardless of small or large, I haven't used data augmentation during training and I think with augmentation we can improve the results further.

We haven't tried other transformer architectures like ViT. From my understanding, with this swin transformer structure, we can extract both local and global features easily. And during decoding, we use parallel self and cross attention, which helps in improving segmentation accuracy by transferring knowledge from encoder to decoder.

Thanks & Regards,
Himashi
